batchflow/models/eager_torch/base.py

```diff
@@ -0,0 +1,1203 @@
+        If `inputs` is specified with all the required shapes, than it serves as size of batch dimension during
+        placeholder (usually np.ndarrays with zeros) creation.
+
+    loss : str, tuple, dict, list
+        Loss function, might be defined in multiple formats.
+
+        If str, then short ``name``.
+        If tuple, then ``(name, *args)``.
+        If dict, then ``{'name': name, **kwargs}``.
+        If list, then sequence of losses in previous formats.
+
+        Name must be one of:
+            - short name (e.g. ``'mse'``, ``'ce'``, ``'l1'``, ``'cos'``, ``'hinge'``,
+              ``'huber'``, ``'logloss'``, ``'dice'``)
+            - a class name from `torch losses <https://pytorch.org/docs/stable/nn.html#loss-functions>`_
+              (e.g. ``'PoissonNLL'`` or ``'TripletMargin'``)
+            - callable
+
+        Examples:
+
+        - ``{'loss': 'mse'}``
+        - ``{'loss': ('KLDiv', {'reduction': 'none'})``
+        - ``{'loss': {'name': MyCustomLoss, 'epsilon': 1e-6}}``
+        - ``{'loss': my_custom_loss_fn}``
+        - ``{'loss': ['dice', 'bce']}``
+
+    optimizer : str, tuple, dict
+        Optimizer, might be defined in multiple formats.
+
+        If str, then short ``name``.
+        If tuple, then ``(name, *args)``.
+        If dict, then ``{'name': name, **kwargs}``.
+
+        Name must be one of:
+            - short name (e.g. ``'Adam'``, ``'Adagrad'``, any optimizer from
+              `torch.optim <https://pytorch.org/docs/stable/optim.html#algorithms>`_)
+            - a class with ``Optimizer`` interface
+            - a callable which takes model parameters and optional args
+
+        Examples:
+
+        - ``{'optimizer': 'Adam'}``
+        - ``{'optimizer': ('SparseAdam', {'lr': 0.01})}``
+        - ``{'optimizer': {'name': 'Adagrad', 'initial_accumulator_value': 0.01}``
+        - ``{'optimizer': {'name': MyCustomOptimizer, momentum=0.95}}``
+
+    decay : str, tuple, dict
+        Learning rate decay algorithm, might be defined in multiple formats.
+        All decays require to have ``n_iters`` as a key in a configuration
+        dictionary that contains the number of iterations in one epoch.
+
+        If str, then short ``name``.
+        If tuple, then ``(name, *args)``.
+        If dict, then ``{'name': name, **kwargs}``.
+
+        Name must be one of:
+
+        - short name (``'exp'``, ``'invtime'``, ``'naturalexp'``, ``'const'``, ``'poly'``)
+        - a class name from `torch.optim.lr_scheduler
+          <https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate>`_
+          (e.g. ``'LambdaLR'``) except ``'ReduceLROnPlateau'``.
+        - a class with ``_LRScheduler`` interface
+        - a callable which takes optimizer and optional args
+
+        Examples:
+
+        - ``{'decay': 'exp'}``
+        - ``{'decay': ('StepLR', {'steps_size': 10000})}``
+        - ``{'decay': {'name': MyCustomDecay, 'decay_rate': .5}``
+
+    n_iters : int
```

Comment:
what about passing this with decay, if these two key always come together. Maybe not now, but in future